vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -1674,114 +1672,167 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
     // "double CCW-wrapping" of the same object.
     type internal SolutionEventsListener(projNode) =
 
-            let mutable waitDialog : IDisposable option = None
+            static let mutable waitDialog : IDisposable option = None
 
-            // During batch active project configuration changes, make sure we only run CSAF once
-            // per batch. Before this change, OnActiveProjectCfgChange was being called twice per
-            // batch per project.
-            let mutable batchState = NonBatch
-            
             // The CCW wrapper seems to prevent an object-identity test, so we determine whether
             // two IVsHierarchy objects are equal by comparing their captions.  (It's ok if this
             // occasionally yields false positives, as this just means we may do a little extra
             // background work.)
-            let GetCaption(hier : IVsHierarchy) =
+            let GetCaption(hier : IVsHierarchy) : string =
                 if hier = null then
-                    null : System.String
+                    null 
                 else
-                    let mutable o : obj = null
-                    let r = hier.GetProperty(VSConstants.VSITEMID_ROOT, int(__VSHPROPID.VSHPROPID_Caption), &o)
-                    if r = VSConstants.S_OK then
-                        o :?> System.String
+                    let mutable o = null
+                    let hr = hier.GetProperty(VSConstants.VSITEMID_ROOT, int(__VSHPROPID.VSHPROPID_Caption), &o)
+                    if hr = VSConstants.S_OK then
+                        o :?> string
                     else
-                        null : System.String
-
-            let OnActiveProjectCfgChange(pIVsHierarchy) =
-                if GetCaption(pIVsHierarchy) = GetCaption(projNode.InteropSafeIVsHierarchy) && batchState <> BatchDone then
-                    projNode.SetProjectFileDirty(projNode.IsProjectFileDirty)
-                    projNode.ComputeSourcesAndFlags()
-                    
-                    if batchState = BatchWaiting then
-                        batchState <- BatchDone
-                VSConstants.S_OK
+                        null 
 
             let UpdateConfig(pHierProj) =
-                // By default, the F# project system keeps its own internal Configuration and Platform in sync with the current active
-                // Configuration and Platform by listening for OnActiveProjectCfgChange events.  However there is one case where the
-                // active cfg changes without an event, and this is during 'Batch Build'.  So we listen for the start and end of 
-                // Batch Build, and manually update the project to the active cfg before/after to set/reset the config.
```

Comment:
Why has this comment been removed? Is this no longer the case? Your other comment didn't lead me to believe that you tried doing a batch build.